tools/buildgen/_mako_renderer.py

```diff
@@ -18,53 +18,61 @@
 """
 
 import getopt
+import glob
```

Comment:
I think the refactoring/improvements to _mako_renderer should be reviewed as a separate PR.
this PR is already pretty complex and like this is unclear if you're actually changing some logic because the codegen changes or if you're just performing cleanup.

If you provided a separate PR that does buildgen cleanup (and provided a good description for that PR), things would probably be quite obvious and such PR would be easy to review. As a side effect of that could also review the existing buildgen code from python3's perspective (which is something that's also needed).